conanfile.py

```diff
@@ -3,86 +3,102 @@
-    archive_sha256 = "549c3120fe8eaaab7f28946e2430fb0d3d4b40b843a5ea52b78dba49795c7e05"
-
-    exports = ["LICENSE.md", "FindDate.cmake"]
+    exports = ["LICENSE.md"]
     exports_sources = ["CMakeLists.txt", "date-config.cmake"]
-
+    generators = "cmake"
     settings = "os", "arch", "compiler", "build_type"
-    requires = (
-        "libcurl/7.56.1@bincrafters/stable",
-        "OpenSSL/1.0.2n@conan/stable",
-        "zlib/1.2.11@conan/stable"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_system_tz_db": [True, False],
+        "use_tz_db_in_dot": [True, False],
+        "disable_string_view": [True, False],
+    }
+    default_options = (
+        "shared=True",
+        "fPIC=True",
+        "use_system_tz_db=True",
+        "use_tz_db_in_dot=False",
+        "disable_string_view=False",
     )
-    options = {"shared": [True, False]}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
 
-    # The folder name when the *.tar.gz release is extracted
-    folder_name = name + "-%s" % src_version
-    # The name of the archive that is downloaded from Github
-    archive_name = "%s.tar.gz" % folder_name
-    # The temporary build diirectory
-    build_dir = "./%s/build" % folder_name
-
-    default_options = ("shared=True",)
-    generators = "cmake"
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
 
     def configure(self):
-        self.requires.add("libcurl/7.56.1@bincrafters/stable", private=False)
-        if self.settings.os == "Macos":
-            self.options["libcurl"].darwin_ssl = False
+        # FIXME: It's not working on Windows
+        if self.settings.os == "Windows":
+            raise Exception("Date is not working on Windows yet.")
 
-    def source(self):
-        tools.download(
-            "{0}/archive/v{1}.tar.gz".format(self.src_url, self.src_version),
-            self.archive_name
-        )
-        tools.check_sha256(
-            self.archive_name,
-            self.archive_sha256
-        )
-        tools.unzip(self.archive_name)
-        os.unlink(self.archive_name)
+    def requirements(self):
+        self.requires("libcurl/7.56.1@bincrafters/stable")
+        self.requires("OpenSSL/1.0.2n@conan/stable")
+        self.requires("zlib/1.2.11@conan/stable")
 
-        os.rename(os.path.join(self.folder_name, "CMakeLists.txt"),
-                  os.path.join(self.folder_name, "CMakeLists_original.txt"))
-        shutil.copy("CMakeLists.txt", os.path.join(self.folder_name, "CMakeLists.txt"))
-        shutil.copy("date-config.cmake", os.path.join(self.folder_name, "date-config.cmake"))
+    def source(self):
+        self.run("git clone https://github.com/HowardHinnant/date.git")
+        with tools.chdir("./{}".format(self.name)):
+            self.run("git checkout 54e8516af223670b75d7a17c2538c6e6d0843c1f .")
+            self._source_subfolder = os.getcwd()
```

Comment:
This line is not executed if the source code has already available (e.g. if the package has been built before with different settings and options). This is causing problems in the Jenkins build, as if `_source_subfolder` is not defined correctly, the _include_ directory is not copied to the created package. My suggestion is to use something like `os.path.join(self.source_folder, "date")` where that variable is needed, instead.